### ISSUE TITLE
Develop a more dynamic interface for Swagmap presentation

### ISSUE THREAD

**Issue by limikael** [tunapanda/swag-old#2]
See [this Discourse post](http://discourse.tunapanda.org/t/swag-interface-mockups-and-brainstorming/46).

**Comment by neerajkamdar** [comment]
Hey guys,

I would like to use the Foundation framework (http://foundation.zurb.com/) for the front end - its open source with MIT license. Does anyone have any objections to using that or would you prefer another front end framework?

Also, do we have a formal folder structure for Swag. Whichever front end framework we end up using, I would like to adhere to the folder structure that suits the project. 

Thanks,
Neeraj

**Comment by usernamenumber** [comment]
Re framework: aside from having started to read a book on Angular once, I have no familiarity with any of these things, so I'm willing to learn anything we can agree on. I've never heard of Foundation before, though, so what's the advantage of it over something like Angular, which seems to be more of a standard (and thus more likely to have people who could come into the project already knowing it). 

Re folder structure: I'll leave that one to @limikael, but I don't think so?

**Comment by neerajkamdar** [comment]
Foundation is a design framework. Makes it very quick to prototype designs. For Swag I was thinking of using Foundation for apps, which uses Angular. See these for a quick overview:  

http://zurb.com/article/1345/design-amazing-single-page-apps-with-the-  

http://www.htmlxprs.com/post/46/what-you-need-to-know-about-zurb-foundation-for-apps

I haven’t played much with Angular myself, so would have to learn that side of foundation. But this is a good excuse to get into it.

**Comment by limikael** [comment]
Hi,

For the front end, do you mean 
1. the thing that manages log in and listing of swagmaps, or 
2. the thing that displays the actual swagmap?

If 1. yeah that sounds good, go ahead!

For 2., I had something in mind where we would use something like PIXI (http://www.pixijs.com/) or Phaser (http://phaser.io/) to get "game like" animations. With "game like" I mean things like this: http://phaser.io/examples/v2/filters/starlight and like this: http://phaser.io/examples/v2/particles/firestarter

However, there is nothing that stops us from having several different viewers for the swagmaps, and we can test them and see which ones users prefer. The thing that would make them work together with the rest of the system would be the json format for the swagmaps, as long as that format is the same then we can have different versions.

So don't feel that there is anything stopping you from starting on something! Since the whole system is quite modular, we can experiment with different version of the components that use different frameworks and stuff. So we don't need to decide on this beforehand I think.

Makes sense?

**Comment by Jakeii** [comment]
@neerajkamdar If you're not tied to Angular, may I suggest Ember.js? I have a good amount of experience with ember, I could be extra useful if we went that way! Will work just as well with foundation as well :)

**Comment by neerajkamdar** [comment]
@Jakeii Definitely. I haven't used Angular before - I was planning on learning it for this, but since you recommend Ember.js and have experience with it to guide us, we could definitely use that. Thanks.

**Comment by usernamenumber** [comment]
Related to this: https://github.com/tunapanda/swagmapviewer/issues/4
